### PR TITLE
(Support) Annotation Manager

### PR DESCRIPTION
**Pull request by dcupidon**
Added the `annotationManagerEnabled`, `userId` and `userName` configuration options.

### CODE REVIEW

File: ios/Classes/PTFlutterDocumentController.m [hunk 1]
```diff
@@ -260,7 +273,7 @@ -(void)toolManager:(PTToolManager*)toolManager willRemoveAnnotation:(nonnull PTA
         [self.plugin documentController:self annotationsChangedWithActionString:annotationsWithActionString];
     }
     
-    NSString* xfdf = [self generateXfdfCommandWithAdded:Nil modified:Nil removed:@[annotation]];
+    NSString* xfdf = self.isAnnotationManagerEnabled ? [self.pdfViewCtrl.externalAnnotManager GetLastXFDF] : [self generateXfdfCommandWithAdded:Nil modified:Nil removed:@[annotation]];
```
Comment (JamesTron) [inline ios/Classes/PTFlutterDocumentController.m:276]: Isn't "will remove" too early to `GetLastXFDF`? Have you tested?

**Review by JamesTron — CHANGES_REQUESTED** [review]

File: ios/Classes/PTFlutterDocumentController.m (hunk 1 above)
Comment (dcupidon) [reply ios/Classes/PTFlutterDocumentController.m]: I checked and you are right, calling `GetLastXFDF` there returns an incorrect XFDF. I have changed the implementation.

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

File: ios/Classes/PTFlutterDocumentController.m [hunk 2]
```diff
@@ -797,6 +830,27 @@ - (BOOL)toolManager:(PTToolManager *)toolManager shouldHandleLinkAnnotation:(PTA
...
 
+#pragma mark - Notification
+
+- (void)undoManagerSentNotification:(NSNotification *) notification
+{
+    if (self.isAnnotationManagerEnabled && self.userId) {
+        NSString* xfdf = [self.pdfViewCtrl.externalAnnotManager GetLastXFDF];
+        NSError* error;
+        
+        [self.toolManager.annotationManager updateAnnotationsWithXFDFCommand:xfdf error:&error];
```
Comment (JamesTron) [inline ios/Classes/PTFlutterDocumentController.m:841]: I'm a bit surprised this is necessary. Please confirm that it is @dluco ?

**Review by JamesTron — COMMENTED** [review]
Please check with @dluco about my comment. Other than that it looks good.

File: ios/Classes/PTFlutterDocumentController.m (hunk 2 above)
Comment (dluco) [reply ios/Classes/PTFlutterDocumentController.m:841]: @JamesTron Sorry, I got mixed up as to where `updateAnnotationsWithXFDF...` would be called - you're correct that it isn't necessary here.
```suggestion
```

**Review by JamesTron — APPROVED** [review]
I think we should have @dluco give this a review, too.

**Review by dluco — APPROVED** [review]

**Pull request merged**